src/common/interfaces/IGateway.sol

```diff
@@ -51,10 +42,25 @@ interface IGateway is IMessageHandler, IMessageSender, IGatewayHandler {
 
     // --- Events ---
     event PrepareMessage(uint16 indexed centrifugeId, PoolId poolId, bytes message);
-    event SendBatch(uint16 indexed centrifugeId, bytes32 batchId, bytes batch, IAdapter adapter);
-    event SendProof(uint16 indexed centrifugeId, bytes32 proofId, bytes32 batchHash, IAdapter adapter);
-    event ProcessBatch(uint16 indexed centrifugeId, bytes32 batchId, bytes batch, IAdapter adapter);
-    event ProcessProof(uint16 indexed centrifugeId, bytes32 proofId, bytes32 batchHash, IAdapter adapter);
+    event SendBatch(
+        uint16 indexed centrifugeId,
+        bytes32 payloadId,
+        bytes batch,
+        IAdapter adapter,
+        bytes32 adapterData,
+        address transactionPayer,
```

Comment:
`adapterData` can be used to link a Centrifuge payload ID to a message ID On the adapter side.

`underpaid` can be used to know which transactions still need a manual payment to the adapter's relayer.